src/helpers/analytics.coffee

```diff
@@ -0,0 +1,64 @@
+_ = require 'underscore'
+Router = require 'react-router'
+{HistoryLocation} = require 'react-router'
+DestinationHelper = require '../helpers/routes-and-destinations'
+
+# generate custom event data for routes
+Events =
+  viewTaskStep: ({courseId}) ->
+    # track all work done on a course
+    Analytics.sendEvent 'Course', 'Work', label: courseId
+
+  viewStudentDashboard: ({courseId}) ->
+    # compare activity between courses
+    Analytics.sendEvent 'Student', 'Dashboard', label: courseId
+
+
+# Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
+Translators =
+
+  dashboard:            ({courseId}) -> "/student/choose-course/#{courseId}"
+  viewTaskStep:         ({courseId}) -> "/student/task-step/#{courseId}"
+  viewGuide:            ({courseId}) -> "/student/performance-forecast/#{courseId}"
+  viewStudentDashboard: ({courseId}) -> "/student/dashboard/#{courseId}"
+  viewPractice:         ({courseId}) -> "/student/practice/#{courseId}"
+
+  calendarByDate:        ({courseId}) -> "/teacher/calendar/#{courseId}"
+  viewTeacherGuide:      ({courseId}) -> "/teacher/performance-forecast/#{courseId}"
+  viewScores:            ({courseId}) -> "/teacher/student-scores/#{courseId}"
+  courseSettings:        ({courseId}) -> "/teacher/roster/#{courseId}"
+  editReading:           ({courseId}) -> "/teacher/assignment/edit/#{courseId}"
+  editHomework:          ({courseId}) -> "/teacher/assignment/edit/#{courseId}"
+  editExternal:          ({courseId}) -> "/teacher/assignment/edit/#{courseId}"
+  createReading:         ({courseId}) -> "/teacher/assignment/create/#{courseId}"
+  createHomework:        ({courseId}) -> "/teacher/assignment/create/#{courseId}"
+  createExternal:        ({courseId}) -> "/teacher/assignment/create/#{courseId}"
+  calendarViewPlanStats: ({courseId}) -> "/teacher/metrics/quick/#{courseId}"
+  reviewTask:            ({courseId}) -> "/teacher/metrics/review/#{courseId}"
+
+GA = undefined
+
+Analytics =
+
+  setTracker: (tracker) -> GA = tracker
+
+  onNavigation: (change, router) ->
+    route  = DestinationHelper.routeFromPath(change.path, router.match)
+    return unless GA and route
```

Comment:
It can be null if the page that's loaded initially doesn't match a route, when the router will display the `Invalid` page.  We probably should send a 'page-not-found' hit here.  I'll do so shortly.

Any links that hit the router are caught here and either cleaned up by the `Translators`  or sent as-is if no Translator is present.   

Links that navigate outside the app, like to the accounts page do not appear here.  For those we'll have to add a `Analytics.sendEvent` call if we want to track them.